File: CsharpProjects/TemplateMethodSolucao/TemplateMethodSolucao/MetodoPagamento/PagamentoDinheiro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TemplateMethodSolucao.GatewayCobranca;

namespace TemplateMethodSolucao.MetodoPagamento
{
    public class PagamentoDinheiro : Pagamento
    {
        public PagamentoDinheiro(double valor, Gateway gateway)
        {
            base.valor = valor;
            base.gateway = gateway;
        }

        public override double calcularDesconto()
        {
            return this.valor * 0.1;
        }
    }
}

File: JavaProjects/TemplateMethodSolucao/src/metodoPagamento/PagamentoDinheiro.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package metodoPagamento;

import gatewayPagamento.Gateway;

/**
 *
 * @author jeffe
 */
public class PagamentoDinheiro extends Pagamento {

    public PagamentoDinheiro(double valor, Gateway gateway) {
        super.valor = valor;
        super.gateway = gateway;
    }

    @Override
    public double calcularDesconto(){
            return super.valor*0.1;
    }
}
